float32[] query
int32 k
---
int32[] ids
float32[] scores
